tests/end_to_end_tests/test_register_dataset_alias.py

```diff
@@ -41,3 +42,24 @@ def test_register_dataset_alias(dummy_file):
         assert i < 1
         assert getattr(r, "dataset.dataset_id") == d_id
         assert getattr(r, "dataset_alias.dataset_id") == d_id
+
+    # Try to reuse alias without supersede.  Should fail
+    a2_id = _insert_alias_entry(datareg.Registrar, "nice_dataset_name", d2_id)
+    assert a2_id is None
+
+    # Try again with supersede
```

Comment:
need to check the dataset superseed column has been updated

Reply:
Adding check in next commit.